styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -732,44 +732,24 @@ private static Instant timestampToInstant(Timestamp ts) {
     return Instant.ofEpochSecond(ts.getSeconds(), ts.getNanos());
   }
 
-  Optional<Resource> getResource(String id) {
-    Entity entity = datastore.get(datastore.newKeyFactory().setKind(KIND_RESOURCE).newKey(id));
+  Optional<Tuple2<String, Long>> getCounterLimit(String id) {
```

Comment:
Avoid using `Tuple`. Create a bespoke value class (using e.g. auto-value or auto-matter) instead.

Reply:
My initial idea was not to introduce another class/concept, but I guess it would be much easier to read the code, also clearer type.